### PR TITLE
Add user to orgs by default

### PR DESCRIPTION
**Pull request by sleepycat**
This commit started it's life as a relatively small feature adding a user to an
org by default to ensure that users has a usable account after logging in, but
ended up with some serious mission creep.

Writing tests for this feature uncovered a lack of isolation between
tests, and tests that mostly failed with `KeyError` or
`'NoneType' object is not subscriptable` errors, as well as test data with
clashing ids.

This change had some far reaching consequences and I basically kept reworking
the tests until the isolation errors went away and I could tell if the "user
orgs" feature had actually broken stuff. If it did, I fixed that too.

### CODE REVIEW

File: api/functions/sign_in_user.py [hunk 1]
```diff
@@ -56,11 +56,13 @@ def sign_in_user(user_name, password):
...
                 user_roles.append(temp_dict)
         else:
+            # XXX: Roles is [] || [""] || [{}]?
+            # why not just []?
             user_roles = ["none"]
         try:
             payload = {
                 "exp": datetime.datetime.utcnow()
-                + datetime.timedelta(days=0, seconds=1800),
+                + datetime.timedelta(days=0, seconds=1800), # XXX: too short
```
[the remark this answers is not in the document]
Comment (sleepycat) [reply api/functions/sign_in_user.py:65]: Ha! We'll have to figure that out. I was finding that was so short that it was actually invalidating my jwts while I was debugging stuff!

File: api/functions/sign_in_user.py [hunk 2]
```diff
@@ -56,11 +56,13 @@ def sign_in_user(user_name, password):
                 }
                 user_roles.append(temp_dict)
         else:
+            # XXX: Roles is [] || [""] || [{}]?
```
[the remark this answers is not in the document]
Comment (sleepycat) [reply api/functions/sign_in_user.py:59]: No worries! Little breadcrumbs leading us back to things to simplify.

File: api/functions/auth_wrappers.py [hunk 3]
```diff
@@ -46,6 +46,8 @@ def check_user_claims(user_claims):
     elif user_claims:
         user_id = user_claims[0]['user_id']
         with app.app_context():
+            # XXX: affiliations should have been eager loaded with joinedload
```
[the remark this answers is not in the document]
Comment (sleepycat) [reply api/functions/auth_wrappers.py:49]: If we did a joinedload and loaded both the user and their affiliations in one shot at the beginning of the request that would be good. But I guess you're getting at the idea that another thread/query could change the state in the db between that inital load and when this gets executed. I'll have to think about that.

File: api/functions/sign_in_user.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nslandolt) [reply api/functions/sign_in_user.py:65]: Oh I know that feeling :joy:

**Pull request merged**